pkg/utils/transform/transform.go

```diff
@@ -109,6 +112,136 @@ func InjectLabel(key, value string, overwritePolicy OverwritePolicy, kinds ...st
 	}
 }
 
+func ReplaceDeploymentImages(prefix string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" {
+			return nil
+		}
+
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		images := imagesFromEnv(prefix)
+		containers := d.Spec.Template.Spec.Containers
+		replaceContainerImages(containers, images)
+		replaceContainersArgsImage(containers, images)
+
+		unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(d)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(unstrObj)
+
+		return nil
+	}
+}
+
+func ReplaceTaskImage(prefix, param string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "ClusterTask" {
+			return nil
+		}
+
+		ct := &v1alpha1.ClusterTask{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, ct)
+		if err != nil {
+			return err
+		}
+
+		images := imagesFromEnv(prefix)
+		steps := ct.Spec.Steps
+		replaceStepsImages(steps, images)
+		params := ct.Spec.Inputs.Params
+		images = imagesFor(param, images)
+		replaceParamsImage(params, images)
+
+		unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(ct)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(unstrObj)
+
+		return nil
+	}
+}
+
+func imagesFromEnv(prefix string) map[string]string {
+	images := map[string]string{}
+	for _, env := range os.Environ() {
+		keyValue := strings.Split(env, "=")
+		key := strings.ToLower(keyValue[0])
+		prefix = strings.ToLower(prefix)
+		if !strings.HasPrefix(key, prefix) {
+			continue
+		}
+
+		name := strings.TrimPrefix(key, prefix)
```

Comment:
For the images map (and while naming ENV vars) should we use the Resource Name as well in the name. 
`name := Resource.Name + strings.TrimPrefix(key, prefix)`

Will there be a scenario where 2 Deployments have the same `arg name` and the `key` gets overwritten

I have same concern for ClusterTask - Step Images.

Reply:
may be this kind of collision is handled already, ignore this comment if that is the case

Reply:
>Will there be a scenario where 2 Deployments have the same arg name and the key gets overwritten

true, It could happen but not the case right now. However, if we its critical then we can address it.

Reply:
sounds good